### DISCUSSION TITLE
Received mail is bounced at SMTP endpoint location

### DISCUSSION THREAD

**Discussion by chirmstream** [postalserver/postal#3080] [Help with using Postal]
Hi all,

I have a mailcow server running on my residential network with port 25 blocked.  For the past few years I have been using Dynu's email store/forward and email backup to get around the blocked port 25.  I have started adding some additional mail servers for a few other projects with different domains.  It's becoming too expensive because I have to purchase a store/forward and email backup from Dynu per  domain.  My end goal is to replace what Dynu has been providing me on two (2) different VPS for redundancy, each with port 25 open.

I have gotten my two VPS setup, each running an almost identical postal server config.  The only difference is the primary server is operating on ``smtp.mydomain.com``, and the secondary server is on ``smtp2.mydomain.com``.  Sending emails from any of my mail servers (over port 26) to my postal server(s) works perfect.  I've setup the firewall to route incoming port 26 to port 25.  Postal receives all the outgoing mail and sends them out correctly.  I don't have any deliverability problems to google, apple, etc.

My DNS settings are as follows:

| Type | Name                 | Content                                         |  Priority  |
| ----- | ------------------ | ------------------------------------ | --------- |
| A      | mail                    | home-mailserver-IP                       | N/A       |
| A      | smtp                   | Primary VPS Postal Server IP         | N/A       |
| A      | smtp2                 | Secondary VPS Postal Server IP     | N/A       |
| MX   |  mydomain.com | smtp.mydomain.com                     | 10         |
| MX   |  mydomain.com | smtp2.mydomain.com                   | 20         |
| MX   |  mydomain.com | mail.mydomain.com                      | 30         |

Postal Config:

    version: 2

    postal.yml

    postal:
        web hostname: smtp.mydomain.com
        web protocol: https
        smtp_hostname: smtp.mydomain.com

    main_db:
        host: 127.0.0.1
        username: root 
        password: ***************
        database: postal

    message_db:
        host: 127.0.0.1
        username: root 
        password: ********************
        prefix: postal 

    smtp_server:
        default_bind_address: "::" 

    dns:
        # Specify the DS records that you have configured. Refer to the documentation al
        # https://github.com/atech/postal/wiki/Domains-a-DNS-Configuration for further
        # information about these.
        mx_ records:
            - mx.smtp.mydomain.com
        spf_include: spf.smtp.mydomain.com
        return _path_domain: rp.smtp.mydomain.com
        route_domain: routes.smtp.mydomain.com
        track_domain: track.smtp.mydomain.com

    smtp:
        # Specify an SMIP server that can be used to send messages from the Postal management
        # system to users. You can configure this to use a Postal mail server once the 
        # your installation has been set up. 
        host: 127.0.0.1
        port: 2525
        username: # Complete when Postal is running and you can 
        password: # generate the credentials within the interface. 
        from_name: Postal 
        from_address: smtp.mydomain.com

My problems occur when receiving mail.  What I want to happen is when someone send me an email at ``name@mydomain.com`` the email is sent to the primary Postal server.  The postal server then routes it to my home mail server at ``mail.mydomain.com``.  I have configured to SMTP endpoint in Postal to use port 125, my home router then routes port 125 to port 25 for my residential mail server.

If my primary Postal server is down it will be delivered to the secondary Postal server running on a different VPS.  Secondary postal server is configured the same as the primary server where the SMTP endpoint send it to my home IP.

If my home mail server is down, then Postal will still receive the mail and retry delivery until the home server is online again.

The issue I am getting is mailcow is bouncing all emails that Postal sends it (emails that someone sends to me at ``name@mydomain.com``).

Here is the exact error I get in mailcow Postfix logs:

    0BE6E441EE38: to=<lkbcld@rp.smtp.mydomain.com>, relay=smtp.mydomain.com[162.XXX.XXX.XX]:26, delay=0.81, 
    delays=0/0/0.53/0.27, dsn=5.0.0, 
    status=bounced (host smtp.mydomain.com[162.XXX.XXX.XX] said: 530 From/Sender name is not valid (in reply to end of DATA 
    command))

I must be misunderstanding something regarding receiving mail in Postal.  I do have some other DNS entries for things like ``mx.smtp.mydomain.com`` and ``rp.smtp.mydomain.com`` but have not had any luck.

Thank you

**Comment by chirmstream** [comment]
I'll also add that sending mail to myself (home mail server > postal > back to home mail server) results in the same error.

**Comment by willpower232** [comment]
At some point in the chain, Postal is interpreting the message as being sent, rather than being incoming so it is looking for the correct domain in its configuration which apparently it does not have.

I am not sure of the way around it, unfortunately Postal is quite specific in how it prefers to be used, rather than a super flexible traditional mail server.

**Comment by chirmstream** [comment]
I'm thinking having Postal and the mail server on the same domain is what's causing issues, and why Postal thinks incoming mail is sent mail.  I'm going to try putting Postal on it's own domain and see how it behaves.

**Comment by chirmstream** [comment]
Well I've got postal running on a separate domain, ``send-smtp.org`` if anyone want's to check my dns settings.  Unfortunately it didn't work, my mail server is still bouncing the received mail with the same error ``530 From/Sender name is not valid``

**Reply by willpower232** [reply]
> in reply to end of DATA command

This is the clue, the domain name you're sending from in your email isn't one in your Postal so it doesn't match. Do you need to try using the sender header?

**Comment by chirmstream** [comment]
This is for received mail though.  It's not coming from my domain.  I tried checking the sender header setting but it didn't change anything.  

I was able to accomplish what I wanted using mailcow-dockerized so I am going to close the discussion.
